index.js

```diff
@@ -128,11 +129,21 @@ export const Link = (props) => {
   return cloneElement(jsx, extraProps);
 };
 
+const flattenChildren = (children) => {
+  return Array.isArray(children)
+    ? children.flatMap((c) =>
+        c.type === Fragment
+          ? flattenChildren(c.props.children)
+          : flattenChildren(c)
+      )
+    : [children];
+};
+
 export const Switch = ({ children, location }) => {
   const { matcher } = useRouter();
   const [originalLocation] = useLocation();
 
-  children = Array.isArray(children) ? children : [children];
+  children = flattenChildren(children);
 
   for (const element of children) {
```

Comment:
```suggestion
  for (const element of flattenChildren(children)) {
```

Reply:
Ah yes indeed :) Shaves 2 bytes off the final build. Applied!